presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -188,6 +224,39 @@ public boolean isEquality()
         {
             return upper == lower;
         }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
```

Comment:
Why not use instanceof? It returns false if o == null:
`if (o instanceof BigIntRange)`